### PR TITLE
Remove order_bys from AggregateExec state

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
While working on another PR, I realized that `Arc<dyn AggregateExpr>` has `order_by` method, and keeps its requirement in its state. Hence, I think `order_bys` member which keeps requirement of each aggregate expression is redundant.

## What changes are included in this PR?
This PR removes `order_bys` member from `AggregateExec` struct.

## Are these changes tested?
Existing tests should work.

### CODE REVIEW

**Review by ozankabak — COMMENTED** [review]
Looks like a nice cleanup to me, but a review from @alamb would be helpful to ensure we are not missing some subtle use case.

File: datafusion/physical-plan/src/aggregates/mod.rs [hunk 1]
```diff
@@ -487,10 +483,10 @@ impl AggregateExec {
         ));
         let original_schema = Arc::new(original_schema);
         // Reset ordering requirement to `None` if aggregator is not order-sensitive
-        order_by_expr = aggr_expr
+        let mut order_by_expr = aggr_expr
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/mod.rs:486]: This is the key observation, right? That the ordering can be derived from the `AggregateExprs` themselves

**Review by alamb — APPROVED** [review]
I agree this looks like a nice cleanup to me. Thank you @mustafasrepo

File: datafusion/physical-plan/src/aggregates/mod.rs (hunk 1 above)
Comment (mustafasrepo) [reply datafusion/physical-plan/src/aggregates/mod.rs:486]: Exactly

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Nice, thank you @mustafasrepo

**Pull request merged**